torch/csrc/api/include/torch/nn/modules/loss.h

```diff
@@ -37,6 +37,82 @@ struct TORCH_API L1LossImpl : public Cloneable<L1LossImpl> {
 /// module storage semantics.
 TORCH_MODULE(L1Loss);
 
+// ============================================================================
+/// The `Kullback-Leibler divergence`_ Loss
+///
+/// KL divergence is a useful distance measure for continuous distributions
+/// and is often useful when performing direct regression over the space of
+/// (discretely sampled) continuous output distributions.
+///
+/// As with :class:`~torch.nn.NLLLoss`, the `input` given is expected to contain
+/// *log-probabilities* and is not restricted to a 2D Tensor.
+/// The targets are given as *probabilities* (i.e. without taking the
+/// logarithm).
+///
+/// This criterion expects a `target` `Tensor` of the same size as the
+/// `input` `Tensor`.
+struct TORCH_API KLDivLossImpl : Module {
+  explicit KLDivLossImpl(const KLDivLossOptions& options_ = {});
+
+  /// Pretty prints the `KLDivLoss` module into the given `stream`.
+  void pretty_print(std::ostream& stream) const override;
+
+  Tensor forward(const Tensor& input, const Tensor& target);
+
+  /// The options with which this `Module` was constructed.
+  KLDivLossOptions options;
+};
+
+/// A `ModuleHolder` subclass for `KLDivLossImpl`.
+/// See the documentation for `KLDivLossImpl` class to learn what methods it
+/// provides, or the documentation for `ModuleHolder` to learn about PyTorch's
+/// module storage semantics.
+TORCH_MODULE(KLDivLoss);
+
+// ============================================================================
+
+/// Creates a criterion that measures the mean squared error (squared L2 norm)
+/// between each element in the input :math:`x` and target :math:`y`.
+struct TORCH_API MSELossImpl : Module {
+  explicit MSELossImpl(const MSELossOptions& options_ = {});
+
+  /// Pretty prints the `MSELoss` module into the given `stream`.
+  void pretty_print(std::ostream& stream) const override;
+
+  Tensor forward(const Tensor& input, const Tensor& target);
+
+  /// The options with which this `Module` was constructed.
+  MSELossOptions options;
+};
+
+/// A `ModuleHolder` subclass for `MSELossImpl`.
+/// See the documentation for `MSELossImpl` class to learn what methods it
+/// provides, or the documentation for `ModuleHolder` to learn about PyTorch's
+/// module storage semantics.
+TORCH_MODULE(MSELoss);
+
+// ============================================================================
+
+/// Creates a criterion that measures the Binary Cross Entropy
+/// between the target and the output.
+struct TORCH_API BCELossImpl : Module {
```

Comment:
I think all `Impl` classes need to subclass from `public Cloneable<ImplName>` and implement the `void reset() override` method, otherwise `module->clone()` won't work on them.